tests/test_ilastik/test_workflows/test_HeadlessPixelClassificationWorkflow.py

```diff
@@ -0,0 +1,178 @@
+
+
+@pytest.fixture
+def sample_projects_dir(tmp_path: Path) -> Path:
+    test_data_path = Path(__file__).parent.parent / "data"
+    sample_projects_zip_path = test_data_path / "test_projects.zip"
+    sample_data_dir_path = test_data_path / "inputdata"
+
+    projects_archive = zipfile.ZipFile(sample_projects_zip_path, mode="r")
+    projects_archive.extractall(path=tmp_path)
+
+    shutil.copytree(sample_data_dir_path, tmp_path / "inputdata")
+
+    return tmp_path
+
+
+@pytest.fixture
+def pixel_classification_ilp_2d3c(sample_projects_dir: Path) -> Path:
+    return sample_projects_dir / "PixelClassification2d3c.ilp"
+
+
+def create_h5(data: numpy.ndarray, axiskeys: str) -> Path:
+    assert len(axiskeys) == len(data.shape)
+    path = tempfile.mkstemp()[1] + ".h5"
+    with h5py.File(path, "w") as f:
+        ds = f.create_dataset("data", data=data)
+        ds.attrs["axistags"] = vigra.defaultAxistags(axiskeys).toJSON()
+
+    return Path(path) / "data"
+
+
+class FailedHeadlessExecutionException(Exception):
+    pass
+
+
+def run_headless_pixel_classification(
+    testdir,
+    *,
+    num_distributed_workers: int = 0,
+    project: Path,
+    raw_data: Path,
+    output_filename_format: str,
+    input_axes: str = "",
+    output_format: str = "hdf5",
+    ignore_training_axistags: bool = False,
+):
+    assert project.exists()
+    assert raw_data.parent.exists()
+
+    ilastik_dot_py = Path(__file__).parent.parent.parent.parent / "ilastik.py"
+    subprocess_args = [
+        "python",
+        str(ilastik_dot_py),
+        "--headless",
+        "--project=" + str(project),
+        "--raw-data=" + str(raw_data),
+        "--output_filename_format=" + str(output_filename_format),
+        "--output_format=" + output_format,
+    ]
+
+    if input_axes:
+        subprocess_args.append("--input-axes=" + input_axes)
+
+    if ignore_training_axistags:
+        subprocess_args.append("--ignore_training_axistags")
+
+    if num_distributed_workers:
+        os.environ["OMPI_ALLOW_RUN_AS_ROOT"] = "1"
+        os.environ["OMPI_ALLOW_RUN_AS_ROOT_CONFIRM"] = "1"
+        subprocess_args = ["mpiexec", "-n", str(num_distributed_workers)] + subprocess_args + ["--distributed"]
+
+    result = testdir.run(*subprocess_args)
+    if result.ret != 0:
+        raise FailedHeadlessExecutionException(
+            "===STDOUT===\n\n" + result.stdout.str() + "\n\n===STDERR===\n\n" + result.stderr.str()
+        )
+
+
+def test_headless_2d3c_with_same_raw_data_axis(testdir, pixel_classification_ilp_2d3c: Path, tmp_path: Path):
+    raw_100x100y3c: Path = create_h5(numpy.random.rand(100, 100, 3), axiskeys="yxc")
+    output_path = tmp_path / "out_100x100y3c.h5"
+    run_headless_pixel_classification(
+        testdir, project=pixel_classification_ilp_2d3c, raw_data=raw_100x100y3c, output_filename_format=str(output_path)
+    )
+
+
+def test_headless_2d3c_with_swizzled_raw_data_axis(testdir, pixel_classification_ilp_2d3c: Path, tmp_path: Path):
```

Comment:
...naming things... a term like "permuted" is maybe more common that "swizzled" (at least I had to look it up)